sqlmesh/core/engine_adapter/spark.py

```diff
@@ -324,21 +323,27 @@ def _get_data_objects(
         # Therefore just doing except Exception for now.
         except Exception:
             return []
-        return [
-            DataObject(
-                catalog=self.get_current_catalog(),
-                # This varies between Spark and Databricks
-                schema=(row.asDict() if not isinstance(row, dict) else row).get("namespace")
-                or row["database"],
-                name=row["tableName"],
-                type=(
-                    DataObjectType.VIEW
-                    if "Type: VIEW" in row["information"]
-                    else DataObjectType.TABLE
-                ),
+        data_objects = []
+        catalog = self.get_current_catalog()
+        for row in results:  # type: ignore
+            row_dict = row.asDict() if not isinstance(row, dict) else row
+            schema = row_dict.get("namespace") or row_dict.get("database")
+            # if schema is not present then that means it is a session temp view and can be skipped
+            if not schema:
```

Comment:
How can this be the case if we run `SHOW TABLE EXTENDED IN {schema_name} LIKE '{pattern}` above?

Reply:
It must include temp objects when you run that command.

Reply:
Yep verified this. Testing out using `isTemporary` field instead of checking for null schema.